Replace no-underscore-variable test with pylint.

mattclay opened the pull request:
##### SUMMARY

Replace no-underscore-variable test with pylint.

This should eliminate false positives and false negatives from the no-underscore-variable test which was based on a fairly simple regex.

The pylint good-names and bad-names lists are the defaults in pylint, except that the underscore has been moved from the good-names list to the bad-names list.

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

sanity tests

mattclay committed:
Replace no-underscore-variable test with pylint.

abadger commented:
+1 from me.  The more that we can get pylint doing instead of our ad hoc scripts, the better :-)

mattclay committed:
Update orphan handling for sanity test docs.

- Include disabled tests in the generated list.
- Mark no-underscore-variable as orphaned.

mattclay committed:
Update command for listing tests.
